tools/topology/sof-hda-generic-idisp.m4

```diff
@@ -90,3 +119,15 @@ PCM_PLAYBACK_ADD(HDMI3, 3, PIPELINE_PCM_4)
 DAI_CONFIG(HDA, 0, 1, iDisp1)
 DAI_CONFIG(HDA, 1, 2, iDisp2)
 DAI_CONFIG(HDA, 2, 3, iDisp3)
+DAI_CONFIG(DMIC, 0, 6, dmic32,
+           DMIC_CONFIG(1, 500000, 4800000, 40, 60, 48000,
+                DMIC_WORD_LENGTH(s32le), DMIC, 0,
+                PDM_CONFIG(DMIC, 0, STEREO_PDM0)))
+DAI_CONFIG(DMIC, 1, 7, dmic16,
+           DMIC_CONFIG(1, 500000, 4800000, 40, 60, 48000,
+                DMIC_WORD_LENGTH(s16le), DMIC, 1,
+                PDM_CONFIG(DMIC, 1, STEREO_PDM0)))
+
+
+VIRTUAL_DAPM_ROUTE_IN(dmic32, DMIC, 0, IN, 0)
```

Comment:
@RanderWang does it actually help at all?
What I noticed in the past was that we really need the virtual_widget() and not the route itself to suppress the errors from the machine driver. Can you please double check?

Reply:
@ranj063 , did you see these error ?  I think it is a regression issue.  And for dmic case, I removed them and found no error message.

skl_hda_dsp_generic: ASoC: no source widget found for iDisp3 Tx
skl_hda_dsp_generic: ASoC: Failed to add route iDisp3 Tx -> direct -> hifi3
skl_hda_dsp_generic: ASoC: no source widget found for iDisp3_out
skl_hda_dsp_generic: ASoC: Failed to add route iDisp3_out -> direct -> iDisp3 Tx
skl_hda_dsp_generic: ASoC: no source widget found for iDisp2 Tx
skl_hda_dsp_generic: ASoC: Failed to add route iDisp2 Tx -> direct -> hifi2
skl_hda_dsp_generic: ASoC: no source widget found for iDisp2_out